src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrForwardFlower.cs

```diff
@@ -413,9 +422,14 @@ .. submodules.Select(s => s.Path).Distinct().Select(VmrPatchHandler.GetExclusion
             reapplyVmrPatches: true,
             lookUpBuilds: true,
             amendReapplyCommit: true,
+            resetToRemoteWhenCloningRepo: ShouldResetBranchToRemoteWhenPreparingRepo(),
             cancellationToken: cancellationToken);
     }
 
     protected override NativePath GetEngCommonPath(NativePath sourceRepo) => sourceRepo / Constants.CommonScriptFilesPath;
     protected override bool TargetRepoIsVmr() => true;
+    // When flowing local repos, we should never reset branches to the remote ones, we might lose some changes devs wanted
+    protected override bool ShouldResetBranchToRemoteWhenPreparingVmr() => false;
+    // In forward flow, we're flowing a specific commit, so we should just check it out, no need to sync local branch to remote
+    protected override bool ShouldResetBranchToRemoteWhenPreparingRepo() => false;
```

Comment:
I think we should not worry too much about the local dev experience. I am pretty sure it will be broken anyway. I think it's easire to just fix it once we implement the server scenarios.